src/api-client.ts

```diff
@@ -43,48 +43,65 @@ export class ApiClient {
   }
 
   async getJobDetails(): Promise<JobDetails> {
-    const detailsURL = `/update_jobs/${this.params.jobId}/details`
+    const detailsURL = `${this.params.dependabotApiUrl}/update_jobs/${this.params.jobId}/details`
     try {
-      const res: any = await this.client.get(detailsURL, {
-        headers: {Authorization: this.params.jobToken}
-      })
-      if (res.status !== 200) {
+      const res = await this.getJsonWithRetry<any>(
+        detailsURL,
+        this.params.jobToken
+      )
+      if (res.statusCode !== 200) {
         throw new JobDetailsFetchingError(
-          `fetching job details: unexpected status code: ${res.status}`
+          `fetching job details: unexpected status code: ${
+            res.statusCode
+          }: ${JSON.stringify(res.result)}`
+        )
+      }
+      if (res.result == null) {
+        throw new JobDetailsFetchingError(
+          `fetching job details: missing response`
         )
       }
 
-      return res.data.data.attributes
+      return res.result.data.attributes
     } catch (error: unknown) {
-      if (axios.isAxiosError(error)) {
-        const err = error
-        if (err.response) {
-          throw new JobDetailsFetchingError(
-            `fetching job details: received code ${err.response
-              ?.status}: ${JSON.stringify(err.response?.data)}`
-          )
-        } else {
-          throw new JobDetailsFetchingError(
-            `fetching job details: ${err.message}`
-          )
-        }
-      } else {
+      if (error instanceof JobDetailsFetchingError) {
+        throw error
+      } else if (error instanceof httpClient.HttpClientError) {
         throw new JobDetailsFetchingError(
-          `fetching job details: ${(error as Error).message}`
+          `fetching job details: unexpected status code: ${error.statusCode}: ${error.message}`
+        )
+      } else if (error instanceof Error) {
+        throw new JobDetailsFetchingError(
+          `fetching job details: ${error.name}: ${error.message}`
         )
       }
+      throw error
```

Comment:
consider adding semicolons at the end to avoid weird JS automatic semicolon injection bugs

Reply:
Hm, our formatter does not allow that